data_validation/src/spark_rapids_validation_tool/validation_scripts/metadata_validation.py

```diff
@@ -0,0 +1,289 @@
+    |      col1|      |      |12.000|11.000|6.090909090909091|              6.0|  3.477198454346414|    3.3166247903554|           |           |
+    |      col3|      |      |      |      |5.090909090909091|5.181818181818182| 3.1766191290283903|  3.060005941764879|           |           |
+    |      col8|12.340|12.330|      |      |       12.4345455|       12.4336364|0.30031195901474267|0.30064173786328213|          3|          4|
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
+    """
+    table1_DF = load_table(spark, format, t1, t1p, pk, e, i, f, "")
+    table2_DF = load_table(spark, format, t2, t2p, pk, e, i, f, "")
+
+    table_metric_df1 = generate_metric_df(spark, table1_DF, i, e, t1)
+    table_metric_df2 = generate_metric_df(spark, table2_DF, i, e, t2)
+    joined_table = table_metric_df1.alias("t1").join(table_metric_df2.alias("t2"), ["ColumnName"])
+
+    cond = (round("t1.min" + t1, p) != round("t2.min" + t2, p)) | \
+           (round("t1.max" + t1, p) != round("t2.max" + t2, p)) | \
+           (round("t1.avg" + t1, p) != round("t2.avg" + t2, p)) | \
+           (round("t1.stddev" + t1, p) != round("t2.stddev" + t2, p)) | \
+           (round("t1.countDistinct" + t1, p) != round("t2.countDistinct" + t2, p))
+
+    # apply condition on the joined table, return the final dataframe
+    result_table = joined_table.select("ColumnName",
+                                       when(col("t1.min"+t1) != col("t2.min"+t2), col("t1.min"+t1)).otherwise('').alias("min_A"),
+                                       when(col("t1.min"+t1) != col("t2.min"+t2), col("t2.min"+t2)).otherwise('').alias("min_B"),
+                                       when(col("t1.max"+t1) != col("t2.max"+t2), col("t1.max"+t1)).otherwise('').alias("max_A"),
+                                       when(col("t1.max"+t1) != col("t2.max"+t2), col("t2.max"+t2)).otherwise('').alias("max_B"),
+                                       when(col("t1.avg"+t1) != col("t2.avg"+t2), col("t1.avg"+t1)).otherwise('').alias("avg_A"),
+                                       when(col("t1.avg"+t1) != col("t2.avg"+t2), col("t2.avg"+t2)).otherwise('').alias("avg_B"),
+                                       when(col("t1.stddev"+t1) != col("t2.stddev"+t2), col("t1.stddev"+t1)).otherwise('').alias("stddev_A"),
+                                       when(col("t1.stddev"+t1) != col("t2.stddev"+t2), col("t2.stddev"+t2)).otherwise('').alias("stddev_B"),
+                                       when(col("t1.countDistinct"+t1) != col("t2.countDistinct"+t2), col("t1.countDistinct"+t1)).otherwise('').alias("countdist_A"),
+                                       when(col("t1.countDistinct"+t1) != col("t2.countDistinct"+t2), col("t2.countDistinct"+t2)).otherwise('').alias("countdist_B")
+                                       ).where(cond).sort(asc("ColumnName"))
+    return result_table
```

Comment:
Can we add some basic stats so that it could be used for performance testing. I think this query is going to take longer time to complete.